### PR TITLE
packages: add support to apt cache operations

### PR DESCRIPTION
**Pull request by cmatsuoka**
Add infrastructure to retrieve packages using apt.

Packages/repo is planned to be moved into a separate project, where
it will be further refactored. Current code follows the snapcraft
implementation.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by cjp256 — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 PyYAML==5.4.1
+http://archive.ubuntu.com/ubuntu/pool/main/p/python-apt/python-apt_2.1.3ubuntu1.tar.xz; sys_platform == 'linux'
```
Comment (cjp256) [inline requirements.txt:2]: As this really needs to come from the host, I'd just leave it at `python-apt` for now.

Comment (cmatsuoka) [reply requirements.txt:2]: It seems that only a very old version (0.7.8) is available at PyPI, unsure if 2.1.3 available anywhere else.

Comment (cjp256) [reply requirements.txt:2]: Ah yes, i mean that it needs to be `apt install`d.  It must match the host apt version.

Comment (sergiusens) [reply requirements.txt:2]: this is not as easy from a python "library" point of view

Comment (sergiusens) [reply requirements.txt:2]: I'd say we should look into hosting our own wheel

**Review by sergiusens — APPROVED** [review]

**Pull request merged**